### PR TITLE
Auto-switch attached goggles

### LINKED ISSUE #649: Attached goggles can't be auto-switched
Switching the squad's head gear with the shortcut doesn't work, if the goggles are attached to a helmet (which is also possible in the new version).

### PR DESCRIPTION
**Pull request by ja2-derek**
Resolves #649.

Re-worked the goggle auto-switch (Ctrl-N). Now it swaps the goggles as requested, be it in an inventory slot or as a helmet attachment.

### Difference from current implementation

It does not put away the wrong goggle, if there is no appropriate gear. It does not put away the Sun Google at night or Night Goggles in daylight. There is no penalty for wearing the wrong goggles.

### Difference from @wvx implementation 

(See cd8ff3c7a3e63e7004e3351e5135e1c255315bbe, 801fdfb0a44b85607fff0bf099b8bdd715da4793) This is simpler, that it does not require IME stuff, does not consider time of day, sector level, lighting or gas.

## Pseudo-code

  - If wearing GASMASK, return
  - Find the optimal gear to wear:
     - If gear is present in inventory, pick it
     - Else, if gear is present in helmet attachment, detach
     - If not yet found, continue with the next preferred gear
  - Find the currently mounted eye gear
     - If HEADPOS1 is an eye gear, pick it
     - If HEADPOS2 is an eye gear, pick it
     - If not found, find a free head slot
  - If no currently mounted gear and no free slot, return
  - Swap the current gear with the optimal gear found
    - If no suitable gear found, try to unequip
        - If found free pocket, swap 
        - Else, if found free and valid helmet slot, attach
        - Else, do nothing
    - If we detached gear before, attach the gear being swapped out
  - showEquipMessage and DirtyMercPanelInterface

## Test cases

Listing only the meaningful combinations - but I have not run all the cases in an actual game yet:

Case # | Switch direction | Head Inv | Pocket Inv | Helmet Inv | Action
-- | -- | -- | -- | -- | --
1 | Night | NONE | NONE | NONE | Nil
2 | Night | NONE | NIGHTGOOGLE| NONE | Swap goggles (Pocket)
3 | Night | NONE | NONE | NIGHTGOOGLE | Swap goggles (Helmet)
4 | Night | SUNGOGGLE | NONE | NONE | Un-equip
5 | Night | SUNGOGGLE | NIGHTGOGGLE | NONE | Swap goggles (Pocket)
6 | Night | SUNGOGGLE | NONE | NIGHTGOGGLE | Swap goggles (Helmet)
7 | Night | SUNGOGGLE | NIGHTGOGGLE | NIGHTGOGGLE | Swap goggles (Pocket)
8 | Night | SUNGOGGLE | NIGHTGOGGLE | UVGOOGLE | Swap goggles (Helmet)
9 | Night | NIGHTGOGGLE | NONE | SUNGOGGLE | Nil
10 | Night | NIGHTGOGGLE | UVGOOGLE | NONE | Swap goggles (Pocket)
11 | Night | UVGOOGLE | NIGHTGOGGLE |  NONE | Nil
12 | Night | GASMASK | NIGHTGOGGLE | NONE | Nil
13 | Day | SUNGOGGLE | NONE | NONE | Nil
14 | Day | SUNGOGGLE | NIGHTGOGGLE | NONE | Nil
15 | Day | NIGHTGOGGLE | NONE | NONE | Un-equip
16 | Day | NIGHTGOGGLE | NONE | SUNGOGGLE | Swap goggles (Helmet)
17 | Day | NIGHTGOGGLE | SUNGOGGLE | NONE | Swap goggles (Pocket)
18 | Day | NIGHTGOGGLE | SUNGOGGLE | SUNGOGGLE | Swap goggles (Pocket)

### CODE REVIEW

**Comment by lynxlynxlynx** [comment]
Do sun googles really have no bad effect if worn at night? And the same for the night vision?
Even if not, it's odd that we don't switch to nothing on the 4 and 15 case.

File: src/externalized/Soldier.cc [hunk 1]
```diff
@@ -273,118 +274,81 @@ static void showGearRemoveMessage(const SOLDIERTYPE* s, uint16_t usItem)
...
+	switchHeadGear(HEAD_GEARS_DAY, HEAD_GEARS_NIGHT);
+}
 
+void Soldier::putDayHeadGear()
+{
+	switchHeadGear(HEAD_GEARS_NIGHT, HEAD_GEARS_DAY);
+}
 
-	if(isHeadPosition(uvGogglesPos))
+void Soldier::switchHeadGear(std::vector<ITEMDEFINE> fromGears, std::vector<ITEMDEFINE> toGears)
```
Comment (lynxlynxlynx) [inline src/externalized/Soldier.cc:290]: Should be passed by reference or just made take a bool to toggle which way to switch.

**Comment by ja2-derek** [comment]
> Do sun googles really have no bad effect if worn at night? And the same for the night vision?
> Even if not, it's odd that we don't switch to nothing on the 4 and 15 case.

From what I find the code, wearing SUNGOGGLES should be better than naked eyes even  at night. But I agree this does not make sense...

https://github.com/ja2-stracciatella/ja2-stracciatella/blob/44002eee515d05fa3bc586c7ebe7419bb8ac6f49/src/game/Tactical/Weapons.cc#L2337-L2341

**Comment by ja2-derek** [comment]
> Even if not, it's odd that we don't switch to nothing on the 4 and 15 case.

Updated. Please see the latest commits

**Pull request merged**